pkg/install/stack/seed-mla/stack.go

```diff
@@ -659,35 +659,33 @@ func upgradeNodeExporterDaemonSets(
 	// 1: find the old daemonset
 	logger.Info("Backing up old Node Exporter daemonset…")
 
-	daemonsetsList := &unstructured.UnstructuredList{}
-	daemonsetsList.SetGroupVersionKind(schema.GroupVersionKind{
+	daemonset := &unstructured.Unstructured{}
+	daemonset.SetGroupVersionKind(schema.GroupVersionKind{
 		Group:   "apps",
-		Kind:    "DaemonSetList",
+		Kind:    "DaemonSet",
 		Version: "v1",
 	})
 
-	nodeExporterMatcher := ctrlruntimeclient.MatchingLabels{
-		"app.kubernetes.io/name": NodeExporterReleaseName,
-	}
-	if err := kubeClient.List(ctx, daemonsetsList, ctrlruntimeclient.InNamespace(NodeExporterNamespace), nodeExporterMatcher); err != nil {
+	key := types.NamespacedName{Name: NodeExporterReleaseName, Namespace: NodeExporterNamespace}
+
+	if err := kubeClient.Get(ctx, key, daemonset); err != nil {
```

Comment:
What happens if the daemon set doesn't exist?

Reply:
We discussed about the upgrade process internally on how it is implemented in general for all the helm charts. Hope we are aligned. Thanks!

Reply:
Yes, thank you. As discussed, we shouldn't fail if we can't find something we are trying to delete anyway. I would recommend that we fix this logic here, and if the same issue exists in other places, we can handle it later in follow up PRs/issues.